`ok_file_has_content()` makes inappropriate tests/messages

```
% ls -l
total 4.0K
lrwxrwxrwx 1 mih mih 108 Apr 28 08:56 dummy -> .git/annex/objects/XM/0p/MD5E-s7--9a0364b9e99bb480dd25e1f0284c8555/MD5E-s7--9a0364b9e99bb480dd25e1f0284c8555

...

  File "/home/mih/hacking/datalad/git/datalad/tests/utils.py", line 464, in ok_file_has_content
    ok_exists(path)
  File "/home/mih/hacking/datalad/git/datalad/tests/utils.py", line 459, in ok_exists
    assert exists(path), 'path %s does not exist' % path
AssertionError: path /tmp/datalad_temp_test_bf2458zgkvouj6/dummy does not exist
```

It clearly exists, it is just a dead symlink.

--- a/datalad/tests/utils.py
+++ b/datalad/tests/utils.py
@@ -376,7 +376,7 @@ def ok_archives_caches(repopath, n=1, persistent=None):
 
 
 def ok_exists(path):
-    assert exists(path), 'path %s does not exist' % path
+    assert exists(path), 'path %s does not exist (or dangling symlink)' % path
 
 
 def ok_file_has_content(path, content, strip=False, re_=False,

